src/axom/primal/operators/detail/intersect_impl.hpp

```diff
@@ -1687,6 +1688,305 @@ AXOM_HOST_DEVICE bool intersect_plane_tet3d(const Plane<T, 3>& p,
   return caseNumber > 0 && caseNumber < 15;
 }
 
+/*! \brief Determines if a line intersects a bilinear patch.
+ * \param [in] p0 The first corner of the bilinear patch.
+ * \param [in] p1 The second corner in ccw order.
+ * \param [in] p2 The third corner.
+ * \param [in] p3 The fourth corner.
+ * \param [in] line The line to intersect with the bilinear patch.
+ * \param [out] u The u parameter(s) of the intersection point.
+ * \param [out] v The v parameter(s) of the intersection point.
+ * \param [out] t The t parameter(s) of the intersection point.
```

Comment:
There can either be 0, 1, 2, or a line segment of intersections (which this function always reports as an intersection) between a line and a bilinear patch. Theoretically this means this function doesn't need dynamic arrays to store the returned intersection points, but I do it here for convenience since the other functions use it. More generally though, I wonder if there's a useful theoretical bound between lines and arbitrary Bezier patches?